packages/assets-controllers/src/selectors/token-selectors.ts

```diff
@@ -0,0 +1,512 @@
+          ) {
+            continue;
+          }
+
+          const rawBalance =
+            tokenBalances[accountAddress]?.[chainId]?.[tokenAddress];
+
+          if (!rawBalance) {
+            continue;
+          }
+
+          if (!groupAssets[accountGroupId]) {
+            groupAssets[accountGroupId] = {};
+          }
+
+          let groupChainAssets = groupAssets[accountGroupId][chainId];
+          if (!groupChainAssets) {
+            groupChainAssets = [];
+            groupAssets[accountGroupId][chainId] = groupChainAssets;
+          }
+
+          const fiatData = getFiatBalanceForEvmToken(
+            rawBalance,
+            token.decimals,
+            marketData,
+            currencyRates,
+            chainId,
+            tokenAddress,
+          );
+
+          groupChainAssets.push({
+            type: 'evm',
+            assetId: tokenAddress,
+            isNative: false,
+            address: tokenAddress,
+            image: token.image ?? '',
+            name: token.name ?? token.symbol,
+            symbol: token.symbol,
+            decimals: token.decimals,
+            balance: stringifyBalanceWithDecimals(
+              hexToBigInt(rawBalance),
+              token.decimals,
+            ),
+            fiat: fiatData
+              ? {
+                  balance: fiatData.balance,
+                  currency: currentCurrency,
+                  conversionRate: fiatData.conversionRate,
+                }
+              : undefined,
+            chainId,
+          });
+        }
+      }
+    }
+
+    return groupAssets;
+  },
+);
+
+const selectAllMultichainAssets = createAssetListSelector(
+  [
+    selectAccountsToGroupIdMap,
+    (state) => state.accountsAssets,
+    (state) => state.assetsMetadata,
+    (state) => state.balances,
+    (state) => state.conversionRates,
+    (state) => state.currentCurrency,
+  ],
+  (
+    accountsMap,
+    multichainTokens,
+    multichainAssetsMetadata,
+    multichainBalances,
+    multichainConversionRates,
+    currentCurrency,
+  ) => {
+    const groupAssets: AssetsByAccountGroup = {};
+
+    for (const [accountId, accountAssets] of Object.entries(multichainTokens)) {
+      for (const assetId of accountAssets) {
+        // TODO: We need a safe way to extract each part of the id (in case of multiple / characters)
+        const [chainId, asset] = assetId.split('/') as [
+          `${string}:${string}`,
+          `${string}:${string}`,
+        ];
+
+        const accountGroupId = accountsMap[accountId];
+
+        const assetMetadata = multichainAssetsMetadata[assetId];
+        if (!assetMetadata) {
+          continue;
+        }
+
+        if (!groupAssets[accountGroupId]) {
+          groupAssets[accountGroupId] = {};
+        }
+
+        let groupChainAssets = groupAssets[accountGroupId][chainId];
+        if (!groupChainAssets) {
+          groupChainAssets = [];
+          groupAssets[accountGroupId][chainId] = groupChainAssets;
+        }
```

Comment:
Same nullish coalescing assignment for default/fallback vals

```suggestion
        groupAssets[accountGroupId] ??= {};
        const groupChainAssets = groupAssets[accountGroupId][chainId] ?? [];
        groupAssets[accountGroupId][chainId] ??= groupChainAssets;
```